### PR TITLE
LPS-196387, LPS 196388, LPS-197425

### PR DESCRIPTION
**Pull request by ces-totrinh**
Hi @ces-luannguyen @ces-quanhuynh @locpham97 
Please help review this PR and leave your comments.
Tickets:
https://liferay.atlassian.net/browse/LPS-196387.
https://liferay.atlassian.net/browse/LPS-196388.
https://liferay.atlassian.net/browse/LPS-197425.
Notes:
The Document tab will be displayed in a folder's permission model when the folder has at least one document (not in the trash) inside, regardless of whether the documents are direct child documents or not.
When accessing the folder permissions for the first time, the child documents' resource permissions will be initialized if there are not any existing resources.
Previous PR: https://github.com/tototrinh/liferay-portal/pull/74
Thanks,

### CODE REVIEW

**Comment by ces-locpham** [comment]
LGTM @tototrinh

**Review by ces-locpham — APPROVED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java [hunk 1]
```diff
@@ -134,11 +150,102 @@ private long _getClassPK(PortletRequest portletRequest) {
 		return resourcePrimKeys[0];
 	}
 
+	private int _getFileEntriesCount(
```
Comment (ces-quanhuynh) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java:153]: Should this method be named _getFileEntryCountInDirectory, that make its purpose clearer.

Comment (tototrinh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java:153]: I put the `folderId` as a parameter, so it should be descriptive of the function's purpose, which is to get file entries in the folder. Thanks.

**Review by ces-quanhuynh — APPROVED** [review]

**Comment by ces-totrinh** [comment]
@hhtank this passed our review. Please help review it. Thanks.

**Pull request merged**